.github/workflows/codegen-preview.yml

```diff
@@ -64,11 +66,72 @@ jobs:
       if: ${{ github.head_ref != null }}
       with:
         script: |
-          const { DIFF_FILE_NAME } = process.env;
-
           await github.rest.issues.createComment({
             issue_number: context.issue.number,
             owner: context.repo.owner,
             repo: context.repo.repo,
             body: '${{ steps.generate-diff.outputs.bot-message }}'
           })
+
+  generate-doc-preview:
+    runs-on: ubuntu-latest
+    name: Generate rustdoc preview and upload to S3
+    env:
+      AWS_REGION: us-west-2
+      S3_BUCKET_NAME: ${{ secrets.SMITHY_RS_PULL_REQUEST_CDN_S3_BUCKET_NAME }}
+    permissions:
+      id-token: write
+      contents: read
+      pull-requests: write
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      name: Gradle Cache
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+      # JDK is needed to generate code
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ env.java_version }}
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.rust_version }}
+        default: true
+    - name: Generate doc preview
+      # Only generate two of the smallest services since these get huge. One of these must be
+      # STS since aws-config depends on it. STS and Transcribe Streaming were chosen below to stay
+      # small while still representing most features. Combined, they are about 11 MB at time of writing.
+      run: |
+        ./gradlew -Paws.services=+sts,+transcribestreaming :aws:sdk:assemble
+        pushd aws/sdk/build/aws-sdk
+        # Remove example crates from workspace
+        sed -i '/examples/d' Cargo.toml
+        cargo doc --no-deps --all-features
```

Comment:
is there a reason you did use `RUSTDOCFLAGS='--enable-index-page -Zunstable-options'` instead of manually making an index? In either case, we should use nightly (because that's what rustdoc uses) and include `--cfg docrs` in the `RUSTDOCFLAGS` environment variable to mimic how the docs will look

Reply:
I wasn't aware of `--enable-index-page`, although I like David's suggestion of differentiating the crate names that have actual changes in them, which is enabled by generating it ourselves.